### PR TITLE
[core] Support rename tag action and procedure

### PR DESCRIPTION
**Pull request by xuzifu666**
### Purpose
Currently tag can not rename after create，this pr is aim to support it.

Linked issue: close #xxx

### CODE REVIEW

File: docs/content/spark/procedures.md [hunk 1]
```diff
@@ -109,6 +109,18 @@ This section introduce all available spark procedures about paimon.
...
+      <td>
+         Rename tag to new tag name. Arguments:
+            <li>table: the target table identifier. Cannot be empty.</li>
+            <li>tag_name: name of the tag. Cannot be empty.</li>
+            <li>new_tag_name: the new tag name to rename.</li>
+      </td>
+      <td>
+         CALL sys.rename_tag(table => 'default.T', tag_name => 'tag1', new_tag_name => 'tag2')
+      </td>
+    </tr>
```
Comment (LinMingQiang) [inline docs/content/spark/procedures.md:123]: `target_name` will be better ?

File: paimon-core/src/main/java/org/apache/paimon/utils/TagManager.java [hunk 2]
```diff
@@ -144,6 +144,22 @@ public void createTag(
         }
     }
 
+    public void renameTag(String tagName, String newTagName) {
+        try {
+            if (!tagExists(tagName)) {
+                throw new RuntimeException(String.format("tag: %s is not exists", tagName));
+            }
```
Comment (LinMingQiang) [inline paimon-core/src/main/java/org/apache/paimon/utils/TagManager.java:151]: The specified tag name [%s] does not exist ?

File: paimon-flink/paimon-flink-common/src/main/java/org/apache/paimon/flink/action/RenameActionFactory.java [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+
+package org.apache.paimon.flink.action;
+
+import org.apache.flink.api.java.tuple.Tuple3;
+
+import java.util.Map;
+import java.util.Optional;
+
+/** Factory to create {@link RenameActionFactory}. */
+public class RenameActionFactory implements ActionFactory {
```
Comment (LinMingQiang) [inline paimon-flink/paimon-flink-common/src/main/java/org/apache/paimon/flink/action/RenameActionFactory.java:27]: RenameTagActionFactory ?

File: paimon-flink/paimon-flink-common/src/main/java/org/apache/paimon/flink/action/RenameTagAction.java [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+        super(warehouse, databaseName, tableName, catalogConfig);
+        this.tagName = tagName;
+        this.newTagName = newTagName;
+    }
+
+    @Override
+    public void run() throws Exception {
+        if (StringUtils.isEmpty(tagName) || StringUtils.isEmpty(newTagName)) {
+            throw new RuntimeException(String.format("tagName or newTagName can not be empty"));
+        }
```
Comment (LinMingQiang) [inline paimon-flink/paimon-flink-common/src/main/java/org/apache/paimon/flink/action/RenameTagAction.java:46]: String.format("The specified tag name [%s] or target tag name [%s] cannot be empty.",  tagName, targetName)

File: paimon-flink/paimon-flink-common/src/main/java/org/apache/paimon/flink/procedure/RenameTagProcedure.java [hunk 5]
```diff
@@ -0,0 +1,58 @@
...
+                @ArgumentHint(name = "table", type = @DataTypeHint("STRING")),
+                @ArgumentHint(name = "tagName", type = @DataTypeHint("STRING")),
+                @ArgumentHint(name = "newTagName", type = @DataTypeHint("STRING"))
+            })
+    public String[] call(
+            ProcedureContext procedureContext, String tableId, String tagName, String newTagName)
+            throws Catalog.TableNotExistException {
+        Table table = catalog.getTable(Identifier.fromString(tableId));
+        table.renameTag(tagName, newTagName);
+        return new String[] {"Success"};
```
Comment (LinMingQiang) [inline paimon-flink/paimon-flink-common/src/main/java/org/apache/paimon/flink/procedure/RenameTagProcedure.java:51]: return : Rename [%s] to [%s] successfully .

File: docs/content/spark/procedures.md (hunk 1 above)
Comment (xuzifu666) [reply docs/content/spark/procedures.md:123]: done

File: paimon-core/src/main/java/org/apache/paimon/utils/TagManager.java (hunk 2 above)
Comment (xuzifu666) [reply paimon-core/src/main/java/org/apache/paimon/utils/TagManager.java:151]: done

File: paimon-flink/paimon-flink-common/src/main/java/org/apache/paimon/flink/action/RenameActionFactory.java (hunk 3 above)
Comment (xuzifu666) [reply paimon-flink/paimon-flink-common/src/main/java/org/apache/paimon/flink/action/RenameActionFactory.java]: done

File: paimon-flink/paimon-flink-common/src/main/java/org/apache/paimon/flink/action/RenameTagAction.java (hunk 4 above)
Comment (xuzifu666) [reply paimon-flink/paimon-flink-common/src/main/java/org/apache/paimon/flink/action/RenameTagAction.java:46]: done

File: paimon-flink/paimon-flink-common/src/main/java/org/apache/paimon/flink/procedure/RenameTagProcedure.java (hunk 5 above)
Comment (xuzifu666) [reply paimon-flink/paimon-flink-common/src/main/java/org/apache/paimon/flink/procedure/RenameTagProcedure.java:51]: done

**Comment by xuzifu666** [comment]
Thanks for @LinMingQiang review.  Had addressed comments.

**Comment by LinMingQiang** [comment]
+1

File: docs/content/spark/procedures.md [hunk 6]
```diff
@@ -109,6 +109,18 @@ This section introduce all available spark procedures about paimon.
          CALL sys.create_tag_from_timestamp(`table` => 'default.T', `tag` => 'my_tag', `timestamp` => 1724404318750, time_retained => '1 d')
       </td>
     </tr>
+    <tr>
+      <td>rename_tag</td>
+      <td>
+         Rename tag to new tag name. Arguments:
```
Comment (wwj6591812) [inline docs/content/spark/procedures.md:115]: Rename a tag with a new tag name.

Comment (wwj6591812) [inline docs/content/spark/procedures.md:115]: Rename a tag with a new tag name.

File: paimon-flink/paimon-flink-common/src/main/java/org/apache/paimon/flink/action/RenameActionTagFactory.java [hunk 7]
```diff
@@ -0,0 +1,71 @@
...
+
+package org.apache.paimon.flink.action;
+
+import org.apache.flink.api.java.tuple.Tuple3;
+
+import java.util.Map;
+import java.util.Optional;
+
+/** Factory to create {@link RenameActionTagFactory}. */
+public class RenameActionTagFactory implements ActionFactory {
```
Comment (wwj6591812) [inline paimon-flink/paimon-flink-common/src/main/java/org/apache/paimon/flink/action/RenameActionTagFactory.java:27]: RenameTagActionFactory?

File: paimon-flink/paimon-flink-common/src/main/java/org/apache/paimon/flink/action/RenameActionTagFactory.java [hunk 8]
```diff
@@ -0,0 +1,71 @@
...
+
+    public static final String IDENTIFIER = "rename_tag";
```
Comment (wwj6591812) [inline paimon-flink/paimon-flink-common/src/main/java/org/apache/paimon/flink/action/RenameActionTagFactory.java:29]: private

File: paimon-flink/paimon-flink-common/src/main/java/org/apache/paimon/flink/action/RenameActionTagFactory.java [hunk 9]
```diff
@@ -0,0 +1,71 @@
...
+                        tablePath.f2,
+                        catalogConfig,
+                        tagName,
+                        targetTagName);
+        return Optional.of(action);
+    }
+
+    @Override
+    public void printHelp() {
+        System.out.println("Action \"rename_tag\" rename a tag with a new tagname.");
```
Comment (wwj6591812) [inline paimon-flink/paimon-flink-common/src/main/java/org/apache/paimon/flink/action/RenameActionTagFactory.java:62]: Typo: In word 'tagname'

File: paimon-flink/paimon-flink-common/src/main/java/org/apache/paimon/flink/procedure/RenameTagProcedure.java [hunk 10]
```diff
@@ -0,0 +1,59 @@
...
+import org.apache.paimon.catalog.Identifier;
+import org.apache.paimon.table.Table;
+
+import org.apache.flink.table.annotation.ArgumentHint;
+import org.apache.flink.table.annotation.DataTypeHint;
+import org.apache.flink.table.annotation.ProcedureHint;
+import org.apache.flink.table.procedure.ProcedureContext;
+
+/**
+ * Create tag procedure. Usage:
```
Comment (wwj6591812) [inline paimon-flink/paimon-flink-common/src/main/java/org/apache/paimon/flink/procedure/RenameTagProcedure.java:31]: Create tag procedure?

File: paimon-flink/paimon-flink-common/src/main/java/org/apache/paimon/flink/procedure/RenameTagProcedure.java [hunk 11]
```diff
@@ -0,0 +1,59 @@
...
+ * </code></pre>
+ */
+public class RenameTagProcedure extends ProcedureBase {
+    public static final String IDENTIFIER = "rename_tag";
+
+    @ProcedureHint(
+            argument = {
+                @ArgumentHint(name = "table", type = @DataTypeHint("STRING")),
+                @ArgumentHint(name = "tagName", type = @DataTypeHint("STRING")),
+                @ArgumentHint(name = "targetTagName", type = @DataTypeHint("STRING"))
```
Comment (wwj6591812) [inline paimon-flink/paimon-flink-common/src/main/java/org/apache/paimon/flink/procedure/RenameTagProcedure.java:44]: add a empty line

File: paimon-flink/paimon-flink-common/src/main/java/org/apache/paimon/flink/procedure/RenameTagProcedure.java [hunk 12]
```diff
@@ -0,0 +1,59 @@
...
+
+/**
+ * Create tag procedure. Usage:
+ *
+ * <pre><code>
+ *  CALL sys.rename_tag('tableId', 'tagName', 'targetTagName')
+ * </code></pre>
+ */
+public class RenameTagProcedure extends ProcedureBase {
+    public static final String IDENTIFIER = "rename_tag";
```
Comment (wwj6591812) [inline paimon-flink/paimon-flink-common/src/main/java/org/apache/paimon/flink/procedure/RenameTagProcedure.java:38]: private

File: docs/content/spark/procedures.md [hunk 13]
```diff
@@ -109,6 +109,18 @@ This section introduce all available spark procedures about paimon.
...
+            <li>table: the target table identifier. Cannot be empty.</li>
+            <li>tag_name: name of the tag. Cannot be empty.</li>
+            <li>target_tag_name: the new tag name to rename.</li>
```
Comment (wwj6591812) [inline docs/content/spark/procedures.md:118]: target_tag_name: the new tag name to rename. Cannot be empty.

**Comment by wwj6591812** [comment]
Why you need this feature?
What the business scenario need this feature in your company?

**Comment by wwj6591812** [comment]
Change title to :"Support rename tag action and procedure.";

**Comment by xuzifu666** [comment]
Thanks for @wwj6591812 review, had addressed comments.

File: docs/content/spark/procedures.md (hunk 13 above)
Comment (xuzifu666) [reply docs/content/spark/procedures.md]: done

File: paimon-flink/paimon-flink-common/src/main/java/org/apache/paimon/flink/procedure/RenameTagProcedure.java (hunk 12 above)
Comment (xuzifu666) [reply paimon-flink/paimon-flink-common/src/main/java/org/apache/paimon/flink/procedure/RenameTagProcedure.java]: done

File: paimon-flink/paimon-flink-common/src/main/java/org/apache/paimon/flink/procedure/RenameTagProcedure.java (hunk 11 above)
Comment (xuzifu666) [reply paimon-flink/paimon-flink-common/src/main/java/org/apache/paimon/flink/procedure/RenameTagProcedure.java:44]: done

File: paimon-flink/paimon-flink-common/src/main/java/org/apache/paimon/flink/action/RenameActionTagFactory.java (hunk 9 above)
Comment (xuzifu666) [reply paimon-flink/paimon-flink-common/src/main/java/org/apache/paimon/flink/action/RenameActionTagFactory.java]: done

File: paimon-flink/paimon-flink-common/src/main/java/org/apache/paimon/flink/action/RenameActionTagFactory.java (hunk 8 above)
Comment (xuzifu666) [reply paimon-flink/paimon-flink-common/src/main/java/org/apache/paimon/flink/action/RenameActionTagFactory.java]: done

File: paimon-flink/paimon-flink-common/src/main/java/org/apache/paimon/flink/action/RenameActionTagFactory.java (hunk 7 above)
Comment (xuzifu666) [reply paimon-flink/paimon-flink-common/src/main/java/org/apache/paimon/flink/action/RenameActionTagFactory.java]: done

File: docs/content/spark/procedures.md (hunk 6 above)
Comment (xuzifu666) [reply docs/content/spark/procedures.md]: done

Comment (xuzifu666) [reply docs/content/spark/procedures.md]: done

**Comment by wwj6591812** [comment]
+1

**Review by JingsongLi — APPROVED** [review]
+1

**Pull request merged**